src/Hyperion/Extensions/StreamEx.cs

```diff
@@ -190,9 +189,37 @@ public static string ReadString(this Stream stream, DeserializerSession session)
 
             var buffer = session.GetBuffer(length);
 
-            stream.Read(buffer, 0, length);
+            stream.ReadFull(buffer, 0, length);
             var res = StringEx.FromUtf8Bytes(buffer, 0, length);
             return res;
         }
+
+        /// <summary>
+        /// Repeats reading from stream until requested bytes were read.
+        /// Returns with partial result if stream can't provide enough bytes
+        /// Fixes issue: https://github.com/akkadotnet/Hyperion/issues/40
+        /// Reference for allowed partial streams: https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.read?redirectedfrom=MSDN&view=netcore-3.1#System_IO_Stream_Read_System_Byte___System_Int32_System_Int32_
+        /// -> "An implementation is free to return fewer bytes than requested even if the end of the stream has not been reached."
+        /// </summary>
+        public static int ReadFull(this Stream stream, byte[] buffer, int offset, int count)
```

Comment:
This function is guaranteed to work only if the underlying stream is guaranteed to eventually return the whole stream. The original problem will still happen if the underlying stream is corrupted and/or closed mid-way.

Reply:
Yes. If the underlying stream is corrupted or closed we cannot do anything here.
The only purpose of the method is to try to read as many bytes from the maybe partial stream as requested.

Reply:
The original problems is not about  corrupted and/or closed mid-way streams. The original problems is about partial streams like the [Networkstream](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.networkstream?view=netcore-3.1). The Networkstream is not closed/corrupted, it just didn't receive all bytes yet.

Reply:
Add `bool allowPartial = false` method parameter